proto/nmt/proof.proto

```diff
@@ -0,0 +1,22 @@
+syntax="proto3";
```

Comment:
I think the idea is to import the proto file from the nmt repository: https://github.com/celestiaorg/nmt/blob/main/pb/proof.proto instead of copy pasting the definition here

Reply:
I think best way is uploading nmt to BSR

Reply:
uploaded to BSR: https://buf.build/celestia/nmt/docs/v0.22.2

Can you please update this PR to use this?

Reply:
Addressed this, it could be better if make a subdirectory under `buf.build/celestia/nmt`

Reply:
Thanks for addressing this 🙏 

>  it could be better if make a subdirectory under buf.build/celestia/nmt

Can you elaborate what this means? I'm not sure I got you

Reply:
Without subdirectory only could import `proof.proto` directly because `proto` is at the top of repo.

If created a subdirectory `nmt` like [`gogoproto`](https://buf.build/gogo/protobuf/tree/main:gogoproto)

Import statement would be `import "nmt/proof.proto"`

Reply:
agreed 👍 

but the problem that would be a breaking change for the nmt repo. That's why we decided not to do it for now

Reply:
`import "proof.proto"` is normal and very few developers need to notice this change